examples/features/authz/server/main.go

```diff
@@ -197,13 +201,30 @@ func main() {
 		log.Fatalf("Loading credentials: %v", err)
 	}
 
-	// Create an authorization interceptor using a static policy.
-	staticInteceptor, err := authz.NewStatic(authzPolicy)
-	if err != nil {
-		log.Fatalf("Creating a static authz interceptor: %v", err)
+	// Create authorization interceptors according to the authz-option command-line flag.
+	var unaryAuthzInt grpc.UnaryServerInterceptor
+	var streamAuthzInt grpc.StreamServerInterceptor
+	switch *authzOpt {
+	case authzOptStatic:
+		// Create an authorization interceptor using a static policy.
+		staticInt, err := authz.NewStatic(authzPolicy)
+		if err != nil {
+			log.Fatalf("Creating a static authz interceptor: %v", err)
+		}
+		unaryAuthzInt, streamAuthzInt = staticInt.UnaryInterceptor, staticInt.StreamInterceptor
+	case authzOptFileWatcher:
+		// Create an authorization interceptor by watching a policy file.
+		fileWatcherInt, err := authz.NewFileWatcher(data.Path("rbac/policy.json"), 100*time.Millisecond)
+		if err != nil {
+			log.Fatalf("Creating a file watcher authz interceptor: %v", err)
+		}
+		unaryAuthzInt, streamAuthzInt = fileWatcherInt.UnaryInterceptor, fileWatcherInt.StreamInterceptor
+	default:
+		log.Fatalf("Invalid authz option: %s", *authzOpt)
```

Comment:
Since this example server is specifically for authz, let's put this check earlier (after parsing flags), then error log and exit the process if an invalid authz option is specified.

Reply:
Updated.